<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Callback</title>
</head>

<body>
    <div>jquery ajax</div>
    <script src="https://code.jquery.com/jquery-3.5.1.js"
        integrity="sha256-QWo7LDvxbWT2tbbQ97B53yJnYU3WhH/C8ycbRAkjPDc=" crossorigin="anonymous"></script>
    <script>
        function CallAjax() {
            //Promise는 항상 resolve, reject를 인자로 받는다.
            return new Promise(function (resolve, reject) {
                $.ajax({
                    url: 'https://api.hnpwa.com/v0/news/1.json',
                    success: function (data) {
                        resolve(data);
                    }
                });
            });
        }

        function fetchData() {
            //1
            var result = [];

            CallAjax()   // <= new Promise
                .then(function () {
                    console.log("데이터 호출 결과", data);
                    result = data;
                    console.log('함수 결과', result);
                })
            //2
            // $.ajax({
            //     // success는 ajax 요청이 성공 했을 때만 실행 된다.
            //     // success 라이브러리에서 정의한 것임! 일반 함수에서는 없음
            //     // success는 ajax에서만 유효한 콜백함수이다.
            //     url: 'https://api.hnpwa.com/v0/news/1.json',
            //     success: function (data) {
            //         console.log("데이터 호출 결과", data);
            //         result = data;
            //         console.log('함수 결과',result);
            //     }
            // });
            // //3
            //  console.log('함수 결과', result);
        }
        fetchData();


    </script>
</body>

</html>